addons/website_event_booth_sale/static/tests/tours/website_event_booth_sale_pricelists.js

```diff
@@ -0,0 +1,323 @@
+/** @odoo-module **/
+
+import tour from 'web_tour.tour';
+
+tour.register('event_booth_sale_pricelists_different_currencies', {
+    test: true,
+    url: '/event',
+}, [
+    // Init: registering the booth
+    {
+        content: 'Open "Test Event Booths" event',
+        trigger: 'h5.card-title span:contains("Test Event Booths")',
+    },
+    {
+        content: 'Go to "Get A Booth" page',
+        trigger: 'li.nav-item a:has(span:contains("Get A Booth"))',
+    },
+    {
+        content: 'Select the booth',
+        trigger: '.o_wbooth_booths input[name="event_booth_ids"]',
+        run: function () {
+            $('.o_wbooth_booths input[name="event_booth_ids"]:lt(1)').click();
+            },
+    },
+    {
+        content: 'Confirm the booth by clicking the submit button',
+        trigger: 'button.o_wbooth_registration_submit',
+    },
+    {
+        content: 'Fill in your contact information',
+        trigger: 'input[name="contact_name"]',
+        run: function () {
+            $('input[name="contact_name"]').val('John Doe');
+            $('input[name="contact_email"]').val('jdoe@example.com');
+            },
+    },
+    {
+        content: 'Submit your informations',
+        trigger: 'button[type="submit"]',
+    },
+    /*
+    Test Case 1:
+    1) Change pricelist to "EUR With Discount Included"
+    2) Verify price of the booth -> 90.00
+    3) Verify price of the selected booth -> 90
+    4) Verify price in the cart -> 90.00
+    */
+    // Change Pricelist
+    {
+        content: "Go to page Shop",
+        trigger: '.nav-link:contains("Shop")',
+    },
+    {
+        content: "Toggle Pricelist",
+        trigger: "div.o_pricelist_dropdown a[data-toggle=dropdown]",
+        run: 'click',
+    },
+    {
+        content: "Activate Pricelist EUR With Discount Included",
+        trigger: '.dropdown-item:contains("EUR With Discount Included")',
+        run: 'click',
+    },
+    {
+        content: 'Wait for pricelist to load',
+        trigger: '.dropdown-toggle:contains("EUR With Discount Included")',
+        run: function () {},
+    },
+    // Verify price of the booth
+    {
+        content: "Go to page Event",
+        trigger: '.nav-link:contains("Event")',
+    },
+    {
+        content: 'Open "Test Event Booths" event',
+        trigger: 'h5.card-title span:contains("Test Event Booths")',
+    },
+    {
+        content: 'Go to "Get A Booth" page',
+        trigger: 'li.nav-item a:has(span:contains("Get A Booth"))',
+    },
```

Comment:
I think we could avoid duplicating the code many times for same steps (other occurrences throughout the two test classes).
Maybe defining arrays of common steps and reusing these, or via a function that returns all steps depending on the case parameters?

Reply:
I tried to reproduce kind of what is done in point_of_sale but they use a more complex approach. I used a more simplistic one. Maybe, that could be interesting in the future to have a shared approach to simplify tour in general (maybe it exists but I didn't find it)